gammapy/modeling/fit.py

```diff
@@ -721,6 +723,24 @@ def covariance_result(self):
         """Optimize result."""
         return self._covariance_result
 
+    def write(self, filename, overwrite=False):
```

Comment:
I think that writing the `FitResult` means exporting the `Models` and the covariance matrix plus the actual information on the fit result. Not just the string.
This could be a yaml file exactly like `Models.to_yaml()` with additional entries for `nfev`, `success`, `message` etc.